src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -322,6 +324,80 @@ private void ValidateLinkedAssemblies(BindingDiagnosticBag diagnostics, Cancella
             }
         }
 
+        internal void DiscoverInterceptorsIfNeeded()
+        {
+            if (!Volatile.Read(ref DeclaringCompilation.InterceptorsDiscoveryComplete))
+            {
+                DiscoverInterceptors();
+                Volatile.Write(ref DeclaringCompilation.InterceptorsDiscoveryComplete, true);
+            }
+        }
+
+        private void DiscoverInterceptors()
```

Comment:
>DiscoverInterceptors [](http://example.com/codeflow?start=21&length=20)

Consider converting this helper to a local function. This way no one will forget to set ```InterceptorsDiscoveryComplete``` flag after calling the method.